Sources/SPTDataLoader.m

```diff
@@ -188,11 +220,19 @@ - (void)failedResponse:(SPTDataLoaderResponse *)response
     if (![self isRequestExpected:response.request]) {
         return;
     }
-
-    [self executeDelegateBlock: ^{
-        [self.delegate dataLoader:self didReceiveErrorResponse:response];
-    }];
-
+    if (self.delegate) {
+        [self executeDelegateBlock:^{
+            [self.delegate dataLoader:self didReceiveErrorResponse:response];
+        }];
+    } else {
+        [self executeDelegateBlock:^{
+            void (^handler)(SPTDataLoaderResult *) = [self.requestHandlers objectForKey:response.request];
+            handler([SPTDataLoaderResult fromResponse:response kind:SPTDataLoaderResultFailure]);
```

Comment:
Always check for `nil` before invoking a block. If `handler == nil`, this will crash.